x-pack/legacy/plugins/maps/public/layers/styles/vector/components/field_select.js

```diff
@@ -58,6 +73,16 @@ export function FieldSelect({ fields, selectedFieldName, onChange, ...rest }) {
     });
   }
 
+  function renderOption(option, searchValue, contentClassName) {
+    return (
+      <span className={contentClassName}>
+        <FieldIcon type={option.value.type} size="m" useColor />
+        &nbsp;
+        <EuiHighlight search={searchValue}>{option.label}</EuiHighlight>
+      </span>
+    );
+  }
+
   return (
     <EuiComboBox
```

Comment:
pull `groupFieldsByOrigin` up to module-scope too. It's a helper function and just clutters the react-component.

It'd end up looking like:
```
  return (
    <EuiComboBox
      selectedOptions={selectedOption ? [selectedOption] : []}
      options={groupFieldsByOrigin(fields)}
      ....
      renderOption={renderOption}
      {...rest}
    />
  );

```